[Bug] GUI: all tools are inaccessible from the GUI - main menu, tool tree or console not working

## Describe the bug
Trying to open any tool dialog with double click from the Tool tree pane yields the following error in the console:

```
Traceback (most recent call last):
  File "/home/vandreo/software/grass-dev/dist.x86_64-pc-
linux-gnu/gui/wxpython/gui_core/treeview.py", line 67, in
<lambda>

lambda evt: self._emitSignal(evt.GetItem(),
self.selectionChanged),


^
^
  File "/home/vandreo/software/grass-dev/dist.x86_64-pc-
linux-gnu/gui/wxpython/gui_core/treeview.py", line 197, in
_emitSignal

signal.emit(node=node, **kwargs)
  File "/home/vandreo/software/grass-dev/dist.x86_64-pc-
linux-gnu/etc/python/grass/pydispatch/signal.py", line 232,
in emit

dispatcher.send(signal=self, *args, **kwargs)
  File "/home/vandreo/software/grass-dev/dist.x86_64-pc-
linux-gnu/etc/python/grass/pydispatch/dispatcher.py", line
344, in send

response = robustapply.robustApply(


^
^
  File "/home/vandreo/software/grass-dev/dist.x86_64-pc-
linux-gnu/etc/python/grass/pydispatch/robustapply.py", line
61, in robustApply

return receiver(*arguments, **named)


^
^
  File "/home/vandreo/software/grass-dev/dist.x86_64-pc-
linux-gnu/gui/wxpython/gui_core/menu.py", line 345, in
OnItemSelected

self.showNotification.emit(message=label)
  File "/home/vandreo/software/grass-dev/dist.x86_64-pc-
linux-gnu/etc/python/grass/pydispatch/signal.py", line 232,
in emit

dispatcher.send(signal=self, *args, **kwargs)
  File "/home/vandreo/software/grass-dev/dist.x86_64-pc-
linux-gnu/etc/python/grass/pydispatch/dispatcher.py", line
344, in send

response = robustapply.robustApply(


^
^
  File "/home/vandreo/software/grass-dev/dist.x86_64-pc-
linux-gnu/gui/wxpython/main_window/frame.py", line 324, in
SetStatusText

self.statusbar.SetStatusText(*args)
  File "/home/vandreo/software/grass-dev/dist.x86_64-pc-
linux-gnu/gui/wxpython/lmgr/statusbar.py", line 67, in
SetStatusText

self.GetWidget().SetStatusText(*args)
TypeError
:
StatusBar.SetStatusText(): not enough arguments
```

## To reproduce

1. Go to Tools pane in the GUI
2. Double click on any tool, eg. Vector > Topology Maintenance > v.build
3. See error in the Console pane

## System description

```                                                                    
GRASS version: 8.5.0dev                                                         
Code revision: e56c7cddd7                                                       
Build date: 2024-07-30                                                          
Build platform: x86_64-pc-linux-gnu                                             
GDAL: 3.7.1                                                                     
PROJ: 9.2.1                                                                     
GEOS: 3.12.0                                                                    
SQLite: 3.42.0                                                                  
Python: 3.11.6                                                                  
wxPython: 4.2.1                                                                 
Platform: Linux-6.5.0-44-generic-x86_64-with-glibc2.38                          
```

> Can someone check before [e56c7cd](https://github.com/OSGeo/grass/commit/e56c7cddd7a23b612cd491592c4ac355ca1bac61)?

If I understand correctly, just before this commit the GUI was broken.


> Is there a reproduction without opening GUI?

It fails too:```


> r.stats
Traceback (most recent call last):
  File "/usr/local/grass85/gui/wxpython/gui_core/forms.py", line 3254, in OnInit
    self.mf = TaskFrame(
              ^^^^^^^^^^
  File "/usr/local/grass85/gui/wxpython/gui_core/forms.py", line 567, in __init__
    self.goutput.showNotification.connect(self.SetStatusText)
  File "/usr/local/grass85/etc/python/grass/pydispatch/signal.py", line 153, in connect
    dispatcher.connect(receiver=handler, signal=self, weak=weak)
  File "/usr/local/grass85/etc/python/grass/pydispatch/dispatcher.py", line 143, in connect
    receiver = saferef.safeRef(receiver, onDelete=_removeReceiver)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/grass85/etc/python/grass/pydispatch/saferef.py", line 26, in safeRef
    assert hasattr(target, im_func), (
           ^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: safeRef target <built-in method SetStatusText of TaskFrame object at 0x7f23a615cb80> has __self__, but no __func__, don't know how to create reference
OnInit returned false, exiting...
```

--- a/gui/wxpython/animation/frame.py
+++ b/gui/wxpython/animation/frame.py
@@ -353,7 +353,7 @@ class AnimationFrame(wx.Frame):
         if not self.dialogs["preferences"]:
             dlg = PreferencesDialog(parent=self, giface=self._giface)
             self.dialogs["preferences"] = dlg
-            dlg.formatChanged.connect(self.controller.UpdateAnimations)
+            dlg.formatChanged.connect(lambda: self.controller.UpdateAnimations())
             dlg.CenterOnParent()
 
         self.dialogs["preferences"].Show()

--- a/gui/wxpython/gmodeler/dialogs.py
+++ b/gui/wxpython/gmodeler/dialogs.py
@@ -206,7 +206,9 @@ class ModelSearchDialog(wx.Dialog):
             parent=self, giface=giface, menuModel=menuModel.GetModel()
         )
         self.cmd_prompt.promptRunCmd.connect(self.OnCommand)
-        self.cmd_prompt.commandSelected.connect(self.label.SetValue)
+        self.cmd_prompt.commandSelected.connect(
+            lambda command: self.label.SetValue(command)
+        )
         self.search = SearchModuleWidget(
             parent=self.panel, model=menuModel.GetModel(), showTip=True
         )

--- a/gui/wxpython/gmodeler/panels.py
+++ b/gui/wxpython/gmodeler/panels.py
@@ -156,7 +156,9 @@ class ModelerPanel(wx.Panel, MainPageBase):
         self.goutput = GConsoleWindow(
             parent=self, giface=giface, gconsole=self._gconsole
         )
-        self.goutput.showNotification.connect(self.SetStatusText)
+        self.goutput.showNotification.connect(
+            lambda message: self.SetStatusText(message)
+        )
 
         # here events are binded twice
         self._gconsole.Bind(

--- a/gui/wxpython/gui_core/forms.py
+++ b/gui/wxpython/gui_core/forms.py
@@ -564,7 +564,9 @@ class TaskFrame(wx.Frame):
             self._gconsole.mapCreated.connect(self.OnMapCreated)
         self.goutput = self.notebookpanel.goutput
         if self.goutput:
-            self.goutput.showNotification.connect(self.SetStatusText)
+            self.goutput.showNotification.connect(
+                lambda message: self.SetStatusText(message)
+            )
 
         self.notebookpanel.OnUpdateValues = self.updateValuesHook
         guisizer.Add(self.notebookpanel, proportion=1, flag=wx.EXPAND)

--- a/gui/wxpython/gui_core/menu.py
+++ b/gui/wxpython/gui_core/menu.py
@@ -227,7 +227,7 @@ class SearchModuleWindow(wx.Panel):
         self._btnAdvancedSearch.Bind(wx.EVT_BUTTON, lambda evt: self.AdvancedSearch())
 
         self._tree.selectionChanged.connect(self.OnItemSelected)
-        self._tree.itemActivated.connect(self.Run)
+        self._tree.itemActivated.connect(lambda node: self.Run(node))
 
         self._layout()
 

--- a/gui/wxpython/gui_core/prompt.py
+++ b/gui/wxpython/gui_core/prompt.py
@@ -167,9 +167,11 @@ class GPromptSTC(GPrompt, wx.stc.StyledTextCtrl):
         self._loadHistory()
         if giface:
             giface.currentMapsetChanged.connect(self._loadHistory)
-            giface.entryToHistoryAdded.connect(self._addEntryToCmdHistoryBuffer)
+            giface.entryToHistoryAdded.connect(
+                lambda entry: self._addEntryToCmdHistoryBuffer(entry)
+            )
             giface.entryFromHistoryRemoved.connect(
-                self._removeEntryFromCmdHistoryBuffer
+                lambda index: self._removeEntryFromCmdHistoryBuffer(index)
             )
         #
         # bindings
@@ -431,7 +433,7 @@ class GPromptSTC(GPrompt, wx.stc.StyledTextCtrl):
             else:
                 delimiter = char
             parts.append(delimiter + textLeft.rpartition(char)[2])
-        return min(parts, key=len)
+        return min(parts, key=lambda x: len(x))
 
     def ShowList(self):
         """Show sorted auto-completion list if it is not empty"""

--- a/gui/wxpython/gui_core/widgets.py
+++ b/gui/wxpython/gui_core/widgets.py
@@ -1335,7 +1335,7 @@ class SearchModuleWidget(wx.Panel):
             nodes.update(self._model.SearchNodes(key=key, value=value))
 
         nodes = list(nodes)
-        nodes.sort(key=self._model.GetIndexOfNode)
+        nodes.sort(key=lambda node: self._model.GetIndexOfNode(node))
         self._results = nodes
         self._resultIndex = -1
         return sorted([node.data["command"] for node in nodes if node.data["command"]])

--- a/gui/wxpython/history/tree.py
+++ b/gui/wxpython/history/tree.py
@@ -137,8 +137,12 @@ class HistoryBrowserTree(CTreeView):
         self.runIgnoredCmdPattern = Signal("HistoryBrowserTree.runIgnoredCmdPattern")
 
         self._giface.currentMapsetChanged.connect(self.UpdateHistoryModelFromScratch)
-        self._giface.entryToHistoryAdded.connect(self.InsertCommand)
-        self._giface.entryInHistoryUpdated.connect(self.UpdateCommand)
+        self._giface.entryToHistoryAdded.connect(
+            lambda entry: self.InsertCommand(entry)
+        )
+        self._giface.entryInHistoryUpdated.connect(
+            lambda entry: self.UpdateCommand(entry)
+        )
 
         self.SetToolTip(_("Double-click to open the tool"))
         self.selectionChanged.connect(self.OnItemSelected)

--- a/gui/wxpython/iclass/frame.py
+++ b/gui/wxpython/iclass/frame.py
@@ -139,7 +139,9 @@ class IClassMapPanel(DoubleMapPanel):
         # TODO: for vdigit: it does nothing here because areas do not produce
         # this info
         self.firstMapWindow.digitizingInfo.connect(
-            self.statusbarManager.statusbarItems["coordinates"].SetAdditionalInfo
+            lambda text: self.statusbarManager.statusbarItems[
+                "coordinates"
+            ].SetAdditionalInfo(text)
         )
         self.firstMapWindow.digitizingInfoUnavailable.connect(
             lambda: self.statusbarManager.statusbarItems[

--- a/gui/wxpython/lmgr/frame.py
+++ b/gui/wxpython/lmgr/frame.py
@@ -389,7 +389,9 @@ class GMFrame(wx.Frame):
     def _createDataCatalog(self, parent):
         """Initialize Data Catalog widget"""
         self.datacatalog = DataCatalog(parent=parent, giface=self._giface)
-        self.datacatalog.showNotification.connect(self.SetStatusText)
+        self.datacatalog.showNotification.connect(
+            lambda message: self.SetStatusText(message)
+        )
 
     def _createDisplay(self, parent):
         """Initialize Display widget"""
@@ -410,7 +412,9 @@ class GMFrame(wx.Frame):
                 giface=self._giface,
                 model=self._moduleTreeBuilder.GetModel(),
             )
-            self.search.showNotification.connect(self.SetStatusText)
+            self.search.showNotification.connect(
+                lambda message: self.SetStatusText(message)
+            )
         else:
             self.search = None
 
@@ -430,7 +434,9 @@ class GMFrame(wx.Frame):
             menuModel=self._moduleTreeBuilder.GetModel(),
             gcstyle=GC_PROMPT,
         )
-        self.goutput.showNotification.connect(self.SetStatusText)
+        self.goutput.showNotification.connect(
+            lambda message: self.SetStatusText(message)
+        )
 
         self._gconsole.mapCreated.connect(self.OnMapCreated)
         self._gconsole.Bind(
@@ -443,7 +449,9 @@ class GMFrame(wx.Frame):
         """Initialize history browser widget"""
         if not UserSettings.Get(group="manager", key="hideTabs", subkey="history"):
             self.history = HistoryBrowser(parent=parent, giface=self._giface)
-            self.history.showNotification.connect(self.SetStatusText)
+            self.history.showNotification.connect(
+                lambda message: self.SetStatusText(message)
+            )
             self.history.runIgnoredCmdPattern.connect(
                 lambda cmd: self.RunSpecialCmd(command=cmd),
             )
@@ -629,7 +637,9 @@ class GMFrame(wx.Frame):
 
         # add 'console' widget to main notebook page and add connect switch page signal
         self.notebook.AddPage(page=self.goutput, text=_("Console"), name="output")
-        self.goutput.contentChanged.connect(self._switchPage)
+        self.goutput.contentChanged.connect(
+            lambda notification: self._switchPage(notification)
+        )
 
         # add 'history module' widget to main notebook page
         if self.history:

--- a/gui/wxpython/lmgr/layertree.py
+++ b/gui/wxpython/lmgr/layertree.py
@@ -1248,7 +1248,7 @@ class LayerTree(treemixin.DragAndDrop, CT.CustomTreeCtrl):
             lambda value: self.ChangeGroupLayerOpacity(layer=child, value=value)
         )
         # Apply button
-        dlg.applyOpacity.connect(self._recalculateLayerButtonPosition)
+        dlg.applyOpacity.connect(lambda: self._recalculateLayerButtonPosition())
         dlg.CentreOnParent()
 
         if dlg.ShowModal() == wx.ID_OK:
@@ -1288,7 +1288,7 @@ class LayerTree(treemixin.DragAndDrop, CT.CustomTreeCtrl):
             )
         )
         # Apply button
-        dlg.applyOpacity.connect(self._recalculateLayerButtonPosition)
+        dlg.applyOpacity.connect(lambda: self._recalculateLayerButtonPosition())
         dlg.CentreOnParent()
 
         if dlg.ShowModal() == wx.ID_OK:

--- a/gui/wxpython/main_window/frame.py
+++ b/gui/wxpython/main_window/frame.py
@@ -331,7 +331,9 @@ class GMFrame(wx.Frame):
     def _createDataCatalog(self, parent):
         """Initialize Data Catalog widget"""
         self.datacatalog = DataCatalog(parent=parent, giface=self._giface)
-        self.datacatalog.showNotification.connect(self.SetStatusText)
+        self.datacatalog.showNotification.connect(
+            lambda message: self.SetStatusText(message)
+        )
 
     def _createDisplay(self, parent):
         """Initialize Display widget"""
@@ -353,7 +355,9 @@ class GMFrame(wx.Frame):
                 giface=self._giface,
                 model=self._moduleTreeBuilder.GetModel(),
             )
-            self.search.showNotification.connect(self.SetStatusText)
+            self.search.showNotification.connect(
+                lambda message: self.SetStatusText(message)
+            )
         else:
             self.search = None
 
@@ -373,8 +377,12 @@ class GMFrame(wx.Frame):
             menuModel=self._moduleTreeBuilder.GetModel(),
             gcstyle=GC_PROMPT,
         )
-        self.goutput.showNotification.connect(self.SetStatusText)
-        self.goutput.contentChanged.connect(self._focusPage)
+        self.goutput.showNotification.connect(
+            lambda message: self.SetStatusText(message)
+        )
+        self.goutput.contentChanged.connect(
+            lambda notification: self._focusPage(notification)
+        )
 
         self._gconsole.mapCreated.connect(self.OnMapCreated)
         self._gconsole.Bind(
@@ -387,7 +395,9 @@ class GMFrame(wx.Frame):
         """Initialize history browser widget"""
         if not UserSettings.Get(group="manager", key="hideTabs", subkey="history"):
             self.history = HistoryBrowser(parent=parent, giface=self._giface)
-            self.history.showNotification.connect(self.SetStatusText)
+            self.history.showNotification.connect(
+                lambda message: self.SetStatusText(message)
+            )
             self.history.runIgnoredCmdPattern.connect(
                 lambda cmd: self.RunSpecialCmd(command=cmd),
             )

--- a/gui/wxpython/mapdisp/frame.py
+++ b/gui/wxpython/mapdisp/frame.py
@@ -324,7 +324,9 @@ class MapPanel(SingleMapPanel, MainPageBase):
             )
             self._setUpMapWindow(self.MapWindowVDigit)
             self.MapWindowVDigit.digitizingInfo.connect(
-                self.statusbarManager.statusbarItems["coordinates"].SetAdditionalInfo
+                lambda text: self.statusbarManager.statusbarItems[
+                    "coordinates"
+                ].SetAdditionalInfo(text)
             )
             self.MapWindowVDigit.digitizingInfoUnavailable.connect(
                 lambda: self.statusbarManager.statusbarItems[
@@ -364,7 +366,9 @@ class MapPanel(SingleMapPanel, MainPageBase):
             def openATM(selection):
                 self._layerManager.OnShowAttributeTable(None, selection=selection)
 
-            self.toolbars["vdigit"].openATM.connect(openATM)
+            self.toolbars["vdigit"].openATM.connect(
+                lambda selection: openATM(selection)
+            )
             self.Map.layerAdded.connect(self._updateVDigitLayers)
         self.MapWindowVDigit.SetToolbar(self.toolbars["vdigit"])
 
@@ -1243,7 +1247,9 @@ class MapPanel(SingleMapPanel, MainPageBase):
         self.measureController = controller(self._giface, mapWindow=self.GetMapWindow())
         # assure that the mode is ended and lines are cleared whenever other
         # tool is selected
-        self._toolSwitcher.toggleToolChanged.connect(self.measureController.Stop)
+        self._toolSwitcher.toggleToolChanged.connect(
+            lambda: self.measureController.Stop()
+        )
         self.measureController.Start()
 
     def OnProfile(self, event):

--- a/gui/wxpython/mapdisp/main.py
+++ b/gui/wxpython/mapdisp/main.py
@@ -579,8 +579,8 @@ class MapApp(wx.App):
             ),
         )
 
-        self.Map.saveToFile.connect(self.mapDisplay.DOutFile)
-        self.Map.dToRast.connect(self.mapDisplay.DToRast)
+        self.Map.saveToFile.connect(lambda cmd: self.mapDisplay.DOutFile(cmd))
+        self.Map.dToRast.connect(lambda cmd: self.mapDisplay.DToRast(cmd))
         self.Map.query.connect(
             lambda ltype, maps: self.mapDisplay.SetQueryLayersAndActivate(
                 ltype=ltype, maps=maps

--- a/gui/wxpython/mapswipe/frame.py
+++ b/gui/wxpython/mapswipe/frame.py
@@ -85,8 +85,8 @@ class SwipeMapPanel(DoubleMapPanel):
         self.secondMapWindow.mapQueried.connect(self.Query)
 
         # bind tracking cursosr to mirror it
-        self.firstMapWindow.Bind(wx.EVT_MOTION, self.TrackCursor)
-        self.secondMapWindow.Bind(wx.EVT_MOTION, self.TrackCursor)
+        self.firstMapWindow.Bind(wx.EVT_MOTION, lambda evt: self.TrackCursor(evt))
+        self.secondMapWindow.Bind(wx.EVT_MOTION, lambda evt: self.TrackCursor(evt))
 
         self.MapWindow = self.firstMapWindow  # current by default
         self.firstMapWindow.zoomhistory = self.secondMapWindow.zoomhistory
@@ -784,7 +784,9 @@ class SwipeMapPanel(DoubleMapPanel):
         else:
             self._queryDialog = QueryDialog(parent=self, data=result)
             self._queryDialog.Bind(wx.EVT_CLOSE, self._oncloseQueryDialog)
-            self._queryDialog.redirectOutput.connect(self._giface.WriteLog)
+            self._queryDialog.redirectOutput.connect(
+                lambda output: self._giface.WriteLog(output)
+            )
             self._queryDialog.Show()
 
     def _oncloseQueryDialog(self, event):

--- a/gui/wxpython/rdigit/g.gui.rdigit.py
+++ b/gui/wxpython/rdigit/g.gui.rdigit.py
@@ -139,7 +139,7 @@ def main():
                 rdigit.OnMapSelection()
             # use Close instead of QuitRDigit for standalone tool
             self.rdigit.quitDigitizer.disconnect(self.QuitRDigit)
-            self.rdigit.quitDigitizer.connect(self.Close)
+            self.rdigit.quitDigitizer.connect(lambda: self.Close())
 
             # add Map Display panel to Map Display frame
             sizer = wx.BoxSizer(wx.VERTICAL)

--- a/gui/wxpython/vdigit/g.gui.vdigit.py
+++ b/gui/wxpython/vdigit/g.gui.vdigit.py
@@ -104,7 +104,7 @@ def main():
             self.toolbars["vdigit"].StartEditing(mapLayer)
             # use Close instead of QuitVDigit for standalone tool
             self.toolbars["vdigit"].quitDigitizer.disconnect(self.QuitVDigit)
-            self.toolbars["vdigit"].quitDigitizer.connect(self.Close)
+            self.toolbars["vdigit"].quitDigitizer.connect(lambda: self.Close())
 
             # add Map Display panel to Map Display frame
             sizer = wx.BoxSizer(wx.VERTICAL)

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -304,10 +304,12 @@ ignore = [
 # "INT002", # f-string-in-get-text-func-call
 # "INT001", # format-in-get-text-func-call
 # "INT003", # printf-in-get-text-func-call
+# "PLW0108", # unnecessary-lambda
 # Ignore `E402` (import violations) in all `__init__.py` files
 "*/testsuite/**.py" = ["PT009", "PT027"]
 "__init__.py" = ["E402"]
 "general/g.parser/test.py" = ["INT003"]
+"gui/**" = ["PLW0108"] # See https://github.com/OSGeo/grass/issues/4124
 "gui/wxpython/animation/dialogs.py" = ["INT002"]
 "gui/wxpython/animation/temporal_manager.py" = ["INT003"]
 "gui/wxpython/core/debug.py" = ["INT002"]

